pkgs/development/python-modules/generic/catch_conflicts.py

```diff
@@ -0,0 +1,30 @@
+import pkg_resources
+import collections
+import sys
+
+do_abort = False
+packages = collections.defaultdict(list)
+
+for f in sys.path:
+    for req in pkg_resources.find_distributions(f):
+        if req not in packages[req.project_name]:
+            # some exceptions inside buildPythonPackage
+            if req.project_name in ['setuptools', 'pip']:
+                continue
+            packages[req.project_name].append(req)
+
+
+for name, duplicates in packages.items():
+    if len(duplicates) > 1:
+        do_abort = True
+        print("Found duplicated packages in closure for dependency '{}': ".format(name))
```

Comment:
The one I encountered is `ansible`. You usually just want to use it as an executable, in which case you don't want all of its dependencies' pythonpaths showing up in your environment (the paths are just preserved in the wrapped executable).